src/index.js

```diff
@@ -47,6 +48,19 @@ class Dropzone extends React.Component {
     document.body.onfocus = this.onFileDialogCancel;
   }
 
+  componentWillReceiveProps(nextProps) {
+    // If the accept setting has changed, and component is currently in the accept or reject states
+    if (nextProps.accept !== this.props.accept && (this.state.isDragActive || this.state.isDragReject)) {
+      const allFilesAccepted = this.allFilesAccepted(this.draggedFiles, nextProps);
```

Comment:
Can this code duplication be somehow removed? Let's introduce a new method that would handle this for both cases based on arguments.